### PR TITLE
chore: Update android

### PR DESCRIPTION
**Pull request by robertjcolley**
Android build was wildly out of date. This is the culmination of a few days of battling gradle version issues, kotlin version issues, and java version issues. In this commit, viro react compiles with each of the dependencies (renderer, etc.) and creates the bridge properly both in a React Native project and an Expo project.

There's a "hack" in the Expo withViroAndroid.ts file. Expo generates a MainApplication.kt file that doesn't actually add the ViroReact packages to the application.

Todo:
- [x] Continue fixing arrow functions in some class components
- [x] Rebuild virocore and debug crash in VROMathInvertMatrix
- [x] Add code to show a view instead of initializing AR if the device doesn't support AR Core
- [x] ~~Should I use a library like `react-native-builder-bob` to re-scaffold the library?~~ Deferring to later date
- [ ] Deploy release
- [ ] Update starter kits
- [ ] Update installation instructions for Android [here](https://viro-community.readme.io/docs/installation-instructions#android-linking-you-must-do-this---we-do-not-yet-support-auto-linking)

### CODE REVIEW

**Comment by robertjcolley** [comment]
Currently, this branch is crashing for me at:

image

**Comment by Avtrkrb** [comment]
> Currently, this branch is crashing for me on the `viro-test-bed` 3D demo at:
> 
> image

@robertjcolley Kotlin plugins need to be **v1.8.x** or higher v1.6.x is leading to issues. I'm getting similar issues while building my app. Here are the changes I did:

**android/build.gradle**
```
dependencies {
        classpath 'com.android.tools.build:gradle:8.2.0'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.8.21"
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
```

**android/viro_bridge/build.gradle**
```
dependencies {
    androidTestImplementation('androidx.test.espresso:espresso-core:3.1.0', {
        exclude group: 'com.android.support', module: 'support-annotations'
    })
    implementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.8.21'
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.8.21"
    implementation 'org.jetbrains.kotlin:kotlin-stdlib:1.8.21'
    implementation 'org.jetbrains.kotlin:kotlin-stdlib-common:1.8.21'

    implementation 'androidx.appcompat:appcompat:1.0.0'
    testImplementation 'junit:junit:4.12'
    implementation 'com.facebook.react:react-native:+'
    implementation project(':arcore_client')
    implementation project(':gvr_common')
    implementation project(':viro_renderer')
    implementation 'com.google.android.exoplayer:exoplayer:2.19.1'
    implementation 'com.google.protobuf.nano:protobuf-javanano:3.0.0-alpha-7'

    implementation 'androidx.localbroadcastmanager:localbroadcastmanager:1.0.0'
    implementation "com.facebook.fresco:fresco:3.1.3"
    implementation "com.facebook.fresco:imagepipeline:3.1.3"
    implementation "com.facebook.fresco:imagepipeline-okhttp3:3.1.3"
    implementation 'com.facebook.fresco:imagepipeline-native:3.1.3'
    implementation 'com.facebook.fresco:memory-type-ashmem:3.1.3'
    implementation "com.facebook.fresco:ui-common:3.1.3"
    implementation "com.facebook.fresco:middleware:3.1.3"
    implementation "com.facebook.fresco:animated-gif:3.1.3"
    implementation 'com.facebook.soloader:nativeloader:0.10.5'
    implementation 'com.facebook.fresco:fbcore:3.1.3'
    implementation 'com.facebook.fresco:drawee:3.1.3'
    implementation 'com.facebook.fresco:memory-type-native:3.1.3'
    implementation 'com.facebook.fresco:memory-type-java:3.1.3'
    implementation 'com.facebook.fresco:nativeimagefilters:3.1.3'
    implementation 'com.facebook.fresco:nativeimagetranscoder:3.1.3'


    implementation fileTree(include: ['*.jar'], dir: 'libs')
}
```

FYI: I've got these changes in my branch, will open a PR once I'm done testing it out.

**Comment by Avtrkrb** [comment]
@robertjcolley 
I've raised a few draft PRs to [vircore](https://github.com/NativeVision/virocore/pull/137) & [viro](https://github.com/NativeVision/viro/pull/263) that should fix a lot of issues for android. It's still WIP though, will tag you once it's ready.


> Currently, this branch is crashing for me on the `viro-test-bed` 3D demo at:
> 
> image

My changes should also fix some of these errors.

File: components/ViroScene.tsx [hunk 1]
```diff
@@ -174,7 +174,9 @@ export class ViroScene extends ViroBase<Props> {
   render() {
     // Uncomment this line to check for misnamed props
     //checkMisnamedProps("ViroScene", this.props);
-
+    console.log("[ViroScene].render");
+    console.log("[ViroScene].render", this);
+    console.log("[ViroScene].render", this.props);
```
Comment (illlama) [inline components/ViroScene.tsx:179]: @robertjcolley is these `console.log` are necessary?

Comment (robertjcolley) [reply components/ViroScene.tsx:179]: Nope! I'll remove them in the next release I'm doing this week. Sorry about that! There are a few other places I think I put them.

**Pull request merged**